### PR TITLE
290 general activity testing

### PR DESCRIPTION
**Pull request by wbaccinelli**
In this PR the e2e testing for the general activity dialog has been implemented

### CODE REVIEW

File: features/step_definitions/agenda_steps.js [hunk 1]
```diff
@@ -532,3 +532,144 @@ Then('the dialog is concluded', function (callback) {
...
+
+Then('therapist displays the last input', function (callback) {
+  	context.client.query('SELECT * FROM intervention_activities_performed WHERE users_nicedayuid = $1 and intervention_activity_id = $2',
+					     [context.user_id, context.constants.ACTIVITY_ID],
+					     (err, res) => {
+      if (err) {
+        callback('DB reading error ' + err);
+      } else {
+      verifyRasaResponse(res.rows[0]['user_input'], callback);
+	  }
```
Comment (raar1) [inline features/step_definitions/agenda_steps.js:574]:
```suggestion
    }
```

File: features/step_definitions/agenda_steps.js [hunk 2]
```diff
@@ -532,3 +532,144 @@ Then('the dialog is concluded', function (callback) {
...
+Then('the new input is saved on the DB', function (callback) {
+  	context.client.query('SELECT * FROM intervention_activities_performed WHERE users_nicedayuid = $1 and intervention_activity_id = $2',
+					     [context.user_id, context.constants.ACTIVITY_ID],
+					     (err, res) => {
+      if (err) {
+        callback('DB reading error ' + err);
+      } else {
+        assert(res.rows[0]['user_input'] == context.constants.NEW_INPUT);
+        callback();
+	  }
```
Comment (raar1) [inline features/step_definitions/agenda_steps.js:611]:
```suggestion
    }
```

File: features/step_definitions/agenda_steps.js [hunk 3]
```diff
@@ -532,3 +532,144 @@ Then('the dialog is concluded', function (callback) {
...
+  	context.client.query('SELECT * FROM intervention_activities_performed iap join intervention_activity ia on iap.intervention_activity_id = ia.intervention_activity_id WHERE users_nicedayuid = $1 ORDER BY intervention_activities_performed_id DESC LIMIT 1',
+					     [context.user_id],
+					     (err, res) => {
+      if (err) {
+        callback('DB reading error ' + err);
+      } else {
+        assert(res.rows[0]['intervention_activity_title'] == context.constants.NEW_ACTIVITY_TITLE);
+        context.constants.LAST_ACTIVITY_ROW_ID = res.rows[0]['intervention_activities_performed_id'];
+        callback();
+	  }
```
Comment (raar1) [inline features/step_definitions/agenda_steps.js:661]:
```suggestion
    }
```

File: features/step_definitions/agenda_steps.js [hunk 4]
```diff
@@ -532,3 +532,144 @@ Then('the dialog is concluded', function (callback) {
...
+
+Then('clean DB for further iterations', function (callback) {
+  	context.client.query('DELETE FROM intervention_activities_performed WHERE intervention_activities_performed_id = $1',
+					     [context.LAST_ACTIVITY_ROW_ID],
+					     (err, res) => {
+      if (err) {
+        callback('DB reading error ' + err);
+      } else {
+        callback();
+	  }
```
Comment (raar1) [inline features/step_definitions/agenda_steps.js:673]:
```suggestion
    }
```

**Review by raar1 — APPROVED** [review]
OK so this looks good. These tests are quite long (unavoidably) - how flaky is it in practice?

**Comment by wbaccinelli** [comment]
Yes, it is not really resistant to changes to the text. Maybe we can brainstorm about this and see if we find a more stable solution?

**Pull request merged**